public/app/features/api-keys/MigrateToServiceAccountsCard.tsx

```diff
@@ -26,27 +27,41 @@ export const MigrateToServiceAccountsCard = ({ onMigrate, disabled }: Props): JS
   const migrationBoxDesc = <span>Are you sure you want to migrate all API keys to service accounts? {docsLink}</span>;
 
   return (
-    <Alert title="Switch from API keys to service accounts" severity="warning">
-      <div className={styles.text}>
-        We will soon deprecate API keys. Each API key will be migrated into a service account with a token and will
-        continue to work as they were. We encourage you to migrate your API keys to service accounts now. {docsLink}
-      </div>
-      <div className={styles.actionRow}>
-        <Button className={styles.actionButton} onClick={() => setIsModalOpen(true)}>
-          Migrate all service accounts
-        </Button>
-        <ConfirmModal
-          title={'Migrate API keys to service accounts'}
-          isOpen={isModalOpen}
-          body={migrationBoxDesc}
-          confirmText={'Yes, migrate now'}
-          onConfirm={onMigrate}
-          onDismiss={() => setIsModalOpen(false)}
-          confirmVariant="primary"
-          confirmButtonVariant="primary"
-        />
-      </div>
-    </Alert>
+    <>
+      {apikeysCount > 0 && (
+        <Alert title="Switch from API keys to service accounts" severity="warning">
+          <div className={styles.text}>
+            We will soon deprecate API keys. Each API key will be migrated into a service account with a token and will
+            continue to work as they were. We encourage you to migrate your API keys to service accounts now. {docsLink}
+          </div>
+          <div className={styles.actionRow}>
+            <Button className={styles.actionButton} onClick={() => setIsModalOpen(true)}>
+              Migrate all service accounts
+            </Button>
+            <ConfirmModal
+              title={'Migrate API keys to service accounts'}
+              isOpen={isModalOpen}
+              body={migrationBoxDesc}
+              confirmText={'Yes, migrate now'}
+              onConfirm={onMigrate}
+              onDismiss={() => setIsModalOpen(false)}
+              confirmVariant="primary"
+              confirmButtonVariant="primary"
+            />
+          </div>
+        </Alert>
+      )}
+      {apikeysCount === 0 && (
```

Comment:
this is incase the users goes to the apikeys page and we have no apikeys present.